### PR TITLE
EDU-1089 Add script to migrate S3 data for specific scenario

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-1089

### CODE REVIEW

**Comment by ahelmberger** [comment]
Nice :+1:

**Review by ahelmberger — APPROVED** [review]

File: src/s3-helper.js [hunk 1]
```diff
@@ -107,6 +111,31 @@ export async function copyObjectBetweenDifferentS3Accounts({ sourceS3, destinati
   });
 }
 
+export async function changeObjectKey({ sourceS3, sourceBucketName, objectKey, newObjectKey }) {
```
Comment (ahelmberger) [inline src/s3-helper.js:114]: Just a technicality: The function doesn't really change the key (tho old file will still be there, which is probably a good and safe thing for our use case), it just creates a copy. So we could align the function name to something like: `copyObjectWithinSameBucket`. What do you think?

File: src/migrate-s3-data.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  if (errorOccurred) {
+    console.error('CANCELLED BECAUSE OF ERRORS!');
+  } else {
+    console.log('SUCESSFULLY MIGRATED!');
+
+    console.log(`Deleting old objects within S3 on ${sourceEnv.s3BucketName}`);
+    for (const obj of objectsToMigrate) {
+      console.log(`Deleting object ${obj.Key}`);
+      await deleteObject(sourceS3, sourceEnv.s3BucketName, obj.Key);
```
Comment (ahelmberger) [inline src/migrate-s3-data.js:108]: Maybe we should keep them for a few days? :grimacing:

File: src/migrate-s3-data.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    errorOccurred = true;
+    q.remove(() => true);
+  });
+
+  q.push(objectsToMigrate.map((obj, index) => ({ obj, index })));
+
+  await q.drain();
+
+  if (errorOccurred) {
+    console.error('CANCELLED BECAUSE OF ERRORS!');
```
Comment (ahelmberger) [inline src/migrate-s3-data.js:101]: If there was an error, we should not continue to the next lines, where the old objects are deleted.

File: src/migrate-s3-data.js (hunk 2 above)
Comment (adrianaluchian) [reply src/migrate-s3-data.js:108]: Exactly my train of thought 😄

File: src/s3-helper.js (hunk 1 above)
Comment (adrianaluchian) [reply src/s3-helper.js]: That's true, I wanted to be very specific but then the name is not technically correct.
Changed, thanks!

File: src/migrate-s3-data.js (hunk 2 above)
Comment (ahelmberger) [reply src/migrate-s3-data.js:108]: :+1:

**Pull request merged**